controllers/storageoscluster/metrics-exporter_operand.go

```diff
@@ -0,0 +1,153 @@
+package storageoscluster
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/ondat/operator-toolkit/declarative"
+	"github.com/ondat/operator-toolkit/declarative/kubectl"
+	eventv1 "github.com/ondat/operator-toolkit/event/v1"
+	"github.com/ondat/operator-toolkit/operator/v1/operand"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/kustomize/api/filesys"
+)
+
+const (
+	// metricsExporterPackage contains the resource manifests for metrics-exporter operand.
+	metricsExporterPackage = "metrics-exporter"
+)
+
+type MetricsExporterOperand struct {
+	name            string
+	client          client.Client
+	requires        []string
+	requeueStrategy operand.RequeueStrategy
+	fs              filesys.FileSystem
+	kubectlClient   kubectl.KubectlClient
+
+	prometheusCRDsInstalled bool
+	prometheusCRDsLock      chan struct{}
+}
+
+var _ operand.Operand = &MetricsExporterOperand{}
+
+func (c *MetricsExporterOperand) Name() string                                           { return c.name }
+func (c *MetricsExporterOperand) Requires() []string                                     { return c.requires }
+func (c *MetricsExporterOperand) RequeueStrategy() operand.RequeueStrategy               { return c.requeueStrategy }
+func (c *MetricsExporterOperand) PostReady(ctx context.Context, obj client.Object) error { return nil }
+
+// dependenciesFound dictates wether the dependencies are present
+// and thus we can apply the metrics-exporter resources
+// currently we depend on Prometheus CRDs
+func (c *MetricsExporterOperand) dependenciesFound() bool {
+	return c.prometheusCRDsInstalled
+}
+
+func (c *MetricsExporterOperand) updateDependenciesState(currentState bool) {
+	c.prometheusCRDsLock <- struct{}{}
+	defer func() {
```

Comment:
I think we necessary need a defer here. Only 3 lines of code without error handling or branhces

Reply:
I don't get it, do you mean "we don't need" ?

Reply:
```
c.prometheusCRDsLock <- struct{}{}
c.prometheusCRDsInstalled = currentState
<-c.prometheusCRDsLock
```
Does the same by less code and fewer stack :)

Reply:
Got it now. I will update it